gst/droidcodec/gstdroidvdec.c

```diff
@@ -1143,13 +1155,14 @@ gst_droidvdec_finish (GstVideoDecoder * decoder)
   GST_VIDEO_DECODER_STREAM_UNLOCK (decoder);
   /* Now we wait for the codec to signal EOS */
   g_cond_wait (&dec->state_cond, &dec->state_lock);
+  g_mutex_unlock (&dec->state_lock);
   GST_VIDEO_DECODER_STREAM_LOCK (decoder);
+  g_mutex_lock (&dec->state_lock);
```

Comment:
This should probably moved one line down, because the GST_LOG_OBJECT seems to refer to the stream lock acquired by  GST_VIDEO_DECODER_STREAM_LOCK. There's no need to have any statements in between (especially the one that can potentially block for arbitrary long time)